arangod/Cluster/ClusterMethods.cpp

```diff
@@ -2754,4 +2756,813 @@ int fetchEdgesFromEngines(std::string const& dbname,
+        << "DB server " << req.destination << "is missing backup " << backupId;
+      return arangodb::Result(
+        TRI_ERROR_FILE_NOT_FOUND,
+        std::string("no backup with id ") + backupId + " on server " + req.destination);
+    }
+
+    if (resSlice.hasKey("agency")) {
+      plan.add(resSlice.get("agency"));
+    }
+
+  }
+
+  LOG_TOPIC(DEBUG, Logger::HOTBACKUP)
+    << "Have located backup. Replaying plan snapshot. Requesting restore " << backupId;
+
+  return arangodb::Result();
+  
+}
+
+arangodb::Result hotRestoreCoordinator(std::string const& backupId) {
+
+  // 1. Find local backup with id
+  //    - fail if not found
+  // 2. Match db servers 
+  //    - fail if not matching
+  // 3. Check if they have according backup with backupId
+  //    - fail if not
+  // 4. Stop maintenance feature on all db servers
+  // 5. a. Replay agency
+  //    b. Initiate DB server restores
+
+  ClusterInfo* ci = ClusterInfo::instance();
+  std::vector<ServerID> dbServers = ci->getCurrentDBServers();
+  VPackBuilder plan;
+
+  auto result = findLocalBackup(backupId, dbServers, plan);
+  if (!result.ok()) {
+    LOG_TOPIC(ERR, Logger::HOTBACKUP)
+      << "failed to find backup " << backupId << " on all db servers: "
+      << result.errorMessage();
+      return result;    
+  }
+  if (plan.slice() == VPackSlice::noneSlice()) {
+    LOG_TOPIC(ERR, Logger::HOTBACKUP)
+      << "failed to find agency dump for " << backupId << " on any db server: "
+      << result.errorMessage();
+    return result;    
+  }
+
+  // Match my db servers to those in the backups's agency dump
+  std::unordered_map<ServerID, ServerID> matches;
+  result = matchBackupServers(plan.slice(), dbServers, matches);
+  if (!result.ok()) {
+    LOG_TOPIC(ERR, Logger::HOTBACKUP) << "failed to match db servers: " <<
+      result.errorMessage();
+    return result;
+  }
+
+  // Apply matched servers to create new plan, if any matches to be done,
+  // else just take 
+  VPackBuilder newPlan;
+  if (!matches.empty()) {
+    result = applyDBServerMatchesToPlan(plan.slice(), matches, newPlan);
+    if (!result.ok()) {
+      return result;
+    }
+  }
+
+  // Pause maintenance feature everywhere, fail, if not succeeded everywhere 
+  result = controlMaintenanceFeature("pause", backupId, dbServers);
+  if (!result.ok()) {
+    return result;
+  }
+
+  // Enact new plan upon the agency
+  result = (matches.empty()) ?
+    ci->agencyReplan(plan.slice()): ci->agencyReplan(newPlan.slice());
+  if (!result.ok()) {
+    result = controlMaintenanceFeature("proceed", backupId, dbServers);
+    return result;
+  }
+
+  // Now I will have to wait for the plan to trickle down
+  std::this_thread::sleep_for(std::chrono::seconds(5));
+
+  // Restore all db servers
+  result = restoreOnDBServers(backupId, dbServers);
+  if (!result.ok()) {  // This is disaster!
+    return result;
+  }
+
+  // Proceed maintenance
+  result = controlMaintenanceFeature("proceed", backupId, dbServers);
```

Comment:
Not necessary, since the servers all restart.

Reply:
True. Done